code/modules/mob/death.dm

```diff
@@ -96,6 +96,15 @@
 		kill_CH() //We dead... clear any prepared abilities...
 
 	timeofdeath = world.time
+	if (isanimal(src))
+		set_death_time(ANIMAL, world.time)
+	else if (ispAI(src) || isdrone(src))
+		set_death_time(MINISYNTH, world.time)
+	else if (isliving(src))
+		set_death_time(CREW, world.time)//Crew is the fallback
```

Comment:
MB it's worth to replace that "switch" with samemeaning var?

Reply:
I don't understand what you mean

Reply:
ah, no i don't think thats necessary, this logic works well enough. It only needs to exist in one place and there's no performance constraints, so adding a variable would be unnecessary clutter